gorp_go18_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2012 James Cooper. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// Package gorp provides a simple way to marshal Go structs to and from
+// SQL databases.  It uses the database/sql package, and should work with any
+// compliant database/sql driver.
+//
+// Source code and project home:
+// https://github.com/go-gorp/gorp
+//
+
+// +build go1.8
+
+package gorp_test
+
+import (
+	"testing"
+	"time"
+	"context"
+)
+
+func TestWithCanceledContext(t *testing.T) {
+	if _, driver := dialectAndDriver(); driver != "mysql" {
```

Comment:
I would really like to have a blacklist of drivers that *don't* support it, rather than a whitelist of drivers that *do*.  If you're unsure, assume there is support - if it's not failing in travis, we'll deal with it when it becomes a problem.

Reply:
Good call. It seems like mymysql is the only driver that has not added support (I filed it as an issue https://github.com/ziutek/mymysql/issues/138).

Ran into one other problem though. Some sql dialects (namely sqlite) don't support native sleep calls. I checked the test for the sqlite driver and it looks like they have an attempt at testing a canceled context (https://github.com/mattn/go-sqlite3/blob/master/sqlite3_go18_test.go#L113), but I don't think it's as clean as sleep. I just excluded the languages that don't support sleep for now.